### PR TITLE
Witanday Kyanga

### PR DESCRIPTION
**Pull request by Witanday**

### CODE REVIEW

**Comment by Witanday** [comment]
# Client Side Routing w/ React Router v. 4

Using Link and NavLink to navigate to specific routes
Passing Route Parameters
Passing props to components rendered by the Router
Instructions
Fork this repository, then clone your fork.

## Part 1:
- [ ]  Wrap your app with the router.
- [ ] Inside your App file add two routes.
one route for / that loads the MovieList component.
one route that will take an id parameter after/movies/ (ex: /movies/2, /movies/3 where the id is dynamic). This route should load the Movie component.

## Part 2:
- [ ]  Make it so that the card in MovieList is a link, this should direct the user to the /movies/{id of movie here} URL, where :id is the id of the individual movie.
When a user clicks on a movie card they should be taken to /movies/{id of movie here} to see the details for the selected movie.

- [ ]  Add functionality so the Home button on the SavedList component navigates back to home.
You should now be able to navigate back and forth between the individual movies and the home screen.
## Stretch Goals.
If you have completed Parts 1 & 2 feel free to move on to these stretch goals.

- [ ]  Refactor so that our code is DRY.
You may notice that we are using essentially the same exact JSX code in the Movie component and the MovieDetails component in MovieList.js create a new component in MovieCard.js that returns this JSX code. Then remove the old code from Movie and MovieDetails and instead return the new MovieCard component.

- [ ]  Add Save Movie funcitonality.
You will notice there is a 'Saved Movies' component that we are not currently using. In this step you will add the functionality to save a movie. You will need to pass the addToSavedList function to the Movie component. Once you have done that you will need to add a click handler to the save button.

- [ ]  Turn your Saved Movie list into Links.
You will need to uncomment lines 29-39 in Movie.js to complete this. Your list of saved movies should be links to the movie itself. Study and understand what the saveMovie function is doing.

- [ ]  Turn your Saved Movie Links into NavLinks.

File: client/src/Movies/Movie.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ export default class Movie extends Component {
 
   componentDidMount() {
     // change this line to grab the id passed on the URL
-    const id = 1;
+    const id =this.props.match.params.id;
```
Comment (TERR1E) [inline client/src/Movies/Movie.js]:
```suggestion
    const id = this.props.match.params.id;
```
always remember to space properly

File: client/src/Movies/MovieList.js [hunk 2]
```diff
@@ -24,7 +25,10 @@ export default class MovieList extends Component {
     return (
       <div className="movie-list">
         {this.state.movies.map(movie => (
+          <Link to={`/movies/${movie.id}`}>
           <MovieDetails key={movie.id} movie={movie} />
```
Comment (TERR1E) [inline client/src/Movies/MovieList.js]:
```suggestion
              <MovieDetails key={movie.id} movie={movie} />
```
proper indentation is important too

File: client/src/Movies/SavedList.js [hunk 3]
```diff
@@ -12,7 +13,11 @@ export default class SavedList extends Component {
         {this.props.list.map(movie => (
           <span className="saved-movie">{movie.title}</span>
         ))}
-        <div className="home-button">Home</div>
+         <Link exact to="/">
+         <div className="home-button">Home</div>
```
Comment (TERR1E) [inline client/src/Movies/SavedList.js]:
```suggestion
             <div className="home-button">Home</div>
```
again needs proper indentation

**Pull request closed without merging**